lib/WUI/nhttp/file_info.cpp

```diff
@@ -33,6 +35,63 @@ namespace nhttp::printer {
 using namespace handler;
 
 JsonResult FileInfo::DirRenderer::renderState(size_t resume_point, JsonOutput &output, FileInfo::DirState &state) const {
+    if (api == APIVersion::Octoprint)
+        return renderStateOctoprint(resume_point, output, state);
+    else
+        return renderStateV1(resume_point, output, state);
+}
+
+JsonResult FileInfo::DirRenderer::renderStateV1(size_t resume_point, JsonOutput &output, FileInfo::DirState &state) const {
+    struct stat st;
+
+    // Keep the indentation of the JSON in here!
+    // clang-format off
+    JSON_START;
+    JSON_OBJ_START;
+
+        JSON_CONTROL("\"type\":\"FOLDER\"") JSON_COMMA;
+        JSON_FIELD_BOOL("ro", false) JSON_COMMA;
+        // Note: we can only get a timestamp for folders that are not
+        // /usb/ because that is the root of FATFS and stat() don't work on it
+        if (stat(state.filepath, &st) == 0) {
+            JSON_FIELD_INT("m_timestamp", st.st_mtime) JSON_COMMA;
+        }
+        JSON_FIELD_STR("name", state.filename) JSON_COMMA;
+        JSON_FIELD_ARR("children");
+        while (state.dir.get() && (state.ent = readdir(state.dir.get()))) {
+            if (state.ent->d_type == DT_REG and !filename_is_gcode(state.ent->d_name)) {
```

Comment:
Can we get things like links here now? REG means regular file, DIR is directory… are there others?